src/test/lib/HandshakeTest.cpp

```diff
@@ -278,6 +290,11 @@ QuicTestConnect(
                     }
                 }
 
+                if (AsyncTicketValidation) {
+                    CxPlatSleep(1000);
+                    TEST_QUIC_SUCCEEDED(Server->SetCustomTicketValidationResult(SessionResumption == QUIC_TEST_RESUMPTION_ENABLED_ASYNC));
```

Comment:
You need to wait for `ServerAcceptCtx.NewConnectionReady` just like the `if (AsyncConfiguration)` scope above.